tool/user_workflow_test.py

```diff
@@ -0,0 +1,333 @@
+    except Exception as e:
+        print(f'   ERROR deleting rows: {e}')
+    finally:
+        duration_step6 = time.monotonic() - step6_start_time
+        print(f'--- Step 6 Duration: {duration_step6:.2f} seconds ---')
+
+    # --- Step 7: Create View (Filtered, No Image) ---
+    print('\n--- Step 7: Create View (Filtered, No Image) ---')
+    view_name = 'user_workflow_hf_import_test_dir.filtered_no_image_view'
+    filtered_no_image_view = None
+    step7_start_time = time.monotonic()
+    try:
+        view_filter = images_tbl.image_width < 480
+        filtered_table_for_view = images_tbl.where(view_filter).select(
+            images_tbl.file_name, images_tbl.answer, images_tbl.image_width, images_tbl.split_name, images_tbl.is_jpg
+        )
+        print(f"   Creating view '{view_name}' from table filtered where image_width < 480 (excluding image col)...")
+        filtered_no_image_view = pxt.create_view(view_name, filtered_table_for_view, if_exists='replace')
+        print(f"   View '{view_name}' created successfully.")
+    except Exception as e:
+        print(f'   ERROR creating view: {e}')
+        filtered_no_image_view = None
+    finally:
+        duration_step7 = time.monotonic() - step7_start_time
+        print(f'--- Step 7 Duration: {duration_step7:.2f} seconds ---')
+        if filtered_no_image_view is None:
+            print('   Halting test due to failure in Step 7.')
+
+    if filtered_no_image_view is None:
+        print('   Halting test due to failure in Step 7.')
+        return
+
+    # --- Step 8: Query View (Filtered, No Image) ---
+    print('\n--- Step 8: Query View (Filtered, No Image) ---')
+    step8_start_time = time.monotonic()
+    try:
+        view_count = filtered_no_image_view.count()
+        print(f"   Querying view '{view_name}'. Count: {view_count}")
+        print('   View head:')
+        print(filtered_no_image_view.head(3))
+    except Exception as e:
+        print(f'   ERROR querying view: {e}')
+    finally:
+        duration_step8 = time.monotonic() - step8_start_time
+        print(f'--- Step 8 Duration: {duration_step8:.2f} seconds ---')
+
+    # --- Final Summary ---
+    total_duration = time.monotonic() - total_start_time
+    print("\n--- Test finished. Pixeltable resources are in directory: 'user_workflow_hf_import_test_dir' ---")
+
+    # Print Key Test Parameters
+    print('\n--- Test Parameters ---')
+    print(f'Dataset:                  {"lscpku/RefCOCO_rec"}')
+    print('Rows Imported:            All Splits')
+    print(f'Embedding Model:        {"openai/clip-vit-base-patch32"}')
+
+    # Print summary table
+    print('\n--- Execution Time Summary ---')
+    header = (
+        f'| {"Step":<30} | {"Duration (min)":<15} | {"Duration (s)":<15} | '
+        f'{"Duration (ms)":<15} | {"Rows Affected":<15} |'
+    )
+    separator = f'|{"-" * 32}|{"-" * 17}|{"-" * 17}|{"-" * 17}|{"-" * 17}|'
+    print(header)
+    print(separator)
+    print(
```

Comment:
You might consider using something like `tabulate` for the display here, to make the code easier to read + maintain.